server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -155,6 +158,18 @@ public static IndexRequest getIndexWriteRequest(DocWriteRequest docWriteRequest)
 
     @Override
     protected void doExecute(Task task, BulkRequest bulkRequest, ActionListener<BulkResponse> listener) {
+        long indexingBytes = DocWriteRequest.writeSizeInBytes(bulkRequest.requests.stream());
+        final Releasable releasable = writeMemoryLimits.markCoordinatingOperationStarted(indexingBytes);
+        final ActionListener<BulkResponse> releasingListener = ActionListener.runAfter(listener, releasable::close);
+        threadPool.executor(ThreadPool.Names.WRITE).execute(new ActionRunnable<>(releasingListener) {
```

Comment:
worth noting that for users of the REST API (including our client such as beats, logstash, etc.) there's now a chance that they get a ESRejectedExecution at the full bulk level, not only at the bulk shard item level. We'll need to double-check that this is properly handled. I'm also wondering if we want to directly fork to the write thread pool at this point, or whether we want to keep executing on the http thread (as we used to) or transport thread (as we also used to).
Finally, the threadpool change in `IngestService.executeBulkRequest` is no longer necessary if we go with this change here, as we are already on the write thread pool.

Reply:
I backed out the dispatching. I tend to think we do not need to dispatch in the coordinating only case.